test/Riok.Mapperly.Tests/Mapping/ObjectNestedPropertyTest.cs

```diff
@@ -0,0 +1,261 @@
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Diagnostics;
+
+namespace Riok.Mapperly.Tests.Mapping;
+
+public class ObjectNestedPropertyTest
```

Comment:
Nice test cases 👍 

Could you add test cases for the following:
use `MapNestedProperties`...
* to map to a constructor parameter
* to map to a init only property
* in combination with ignored property casing
* in a queryable projection mapping (probably in another test class)
* with a nullable object (`MapNestedProperties("A")` where `A` is nullable)